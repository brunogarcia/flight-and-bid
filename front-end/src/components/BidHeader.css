.BidHeader-flights {
  text-align: center;
}

.BidHeader-flight {
  margin-bottom: 1em;
}

.BidHeader-flight-icon {
  vertical-align: top;
  transform: rotate(90deg);
}


@media only screen and (min-width: 420px) {
  .BidHeader-flight {
    display: inline-block;
  }

  .BidHeader-flight-icon {
    padding: 1em 2em;
  }
}
