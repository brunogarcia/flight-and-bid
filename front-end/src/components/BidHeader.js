import React from 'react';
import Typography from '@material-ui/core/Typography';
import PlaneIcon from '@material-ui/icons/Flight';
import Types from '../utils/types';
import './BidHeader.css';

const BidHeader = ({ journey }) => {
  const {
    departure,
    arrival,
    departureDate,
    arrivalDate,
  } = journey;

  const hourFormat = {
    hour: 'numeric',
    minute: 'numeric',
  };

  const dateFormat = {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  };

  const departureHourNormalized = new Date(departureDate)
    .toLocaleString('en-us', hourFormat);

  const arrivalHourNormalized = new Date(arrivalDate)
    .toLocaleString('en-us', hourFormat);

  const departureDateNormalized = new Date(departureDate)
    .toLocaleString('en-us', dateFormat);

  const arrivalDateNormalized = new Date(arrivalDate)
    .toLocaleString('en-us', dateFormat);

  return (
    <div className="BidHeader-flights">
      <div className="BidHeader-flight">
        <Typography variant="headline">
          {departure} <small>{departureHourNormalized}</small>
        </Typography>
        <Typography component="p">
          {departureDateNormalized}
        </Typography>
      </div>

      <div className="BidHeader-flight BidHeader-flight-icon">
        <PlaneIcon />
      </div>

      <div className="BidHeader-flight">
        <Typography variant="headline">
          {arrival} <small>{arrivalHourNormalized}</small>
        </Typography>
        <Typography component="p">
          {arrivalDateNormalized}
        </Typography>
      </div>
    </div>
  );
};

BidHeader.propTypes = {
  journey: Types.journey.isRequired,
};


export default BidHeader;

