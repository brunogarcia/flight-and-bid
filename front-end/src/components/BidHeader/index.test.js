import React from 'react';
import renderer from 'react-test-renderer';
import BidHeader from './index';

const journey = {
  key: 'test',
  flight: 'test',
  departure: 'test',
  departureDate: 'test',
  arrival: 'test',
  arrivalDate: 'test',
};

test('renders correctly', () => {
  const component = renderer.create(<BidHeader
    journey={journey}
  />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
