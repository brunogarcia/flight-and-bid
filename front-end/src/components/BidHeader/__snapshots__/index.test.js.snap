// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="BidHeader-flights"
>
  <div
    className="BidHeader-flight"
  >
    <h1
      className="MuiTypography-root-1 MuiTypography-headline-6"
    >
      test
       
      <small>
        Invalid Date
      </small>
    </h1>
    <p
      className="MuiTypography-root-1 MuiTypography-body1-10"
    >
      Invalid Date
    </p>
  </div>
  <div
    className="BidHeader-flight BidHeader-flight-icon"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-25"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M10.18 9"
        />
        <path
          d="M21 16v-2l-8-5V3.5c0-.83-.67-1.5-1.5-1.5S10 2.67 10 3.5V9l-8 5v2l8-2.5V19l-2 1.5V22l3.5-1 3.5 1v-1.5L13 19v-5.5l8 2.5z"
        />
      </g>
    </svg>
  </div>
  <div
    className="BidHeader-flight"
  >
    <h1
      className="MuiTypography-root-1 MuiTypography-headline-6"
    >
      test
       
      <small>
        Invalid Date
      </small>
    </h1>
    <p
      className="MuiTypography-root-1 MuiTypography-body1-10"
    >
      Invalid Date
    </p>
  </div>
</div>
`;
